src/sentry/rules/processing/delayed_processing.py

```diff
@@ -1,36 +1,227 @@
+            project=project, data=condition_data, rule=rule
+        )
+        if not isinstance(condition_inst, EventCondition):
+            logger.warning("Unregistered condition %r", condition_cls.id)
+            return None
+
+        _, duration = condition_inst.intervals[unique_condition.interval]
+        end = datetime.now()
+        # For conditions with interval >= 1 hour we don't need to worry about read your writes
+        # consistency. Disable it so that we can scale to more nodes.
+        option_override_cm: contextlib.AbstractContextManager[object] = contextlib.nullcontext()
+        if duration >= timedelta(hours=1):
+            option_override_cm = options_override({"consistent": False})
+
+        with option_override_cm:
+            start = end - duration
+            results = condition_inst.batch_query(
+                group_ids=group_ids,
+                start=start.replace(tzinfo=UTC),
+                end=end.replace(tzinfo=UTC),
+                environment_id=unique_condition.environment_id,
+            )
+
+            # If the condition is a percent comparison, we need to query the
+            # previous interval to compare against the current interval
+            if condition_data:
+                comparison_type = condition_data.get("comparisonType", ComparisonType.COUNT)
+                if comparison_type == ComparisonType.PERCENT:
+                    comparison_interval = condition_inst.intervals[unique_condition.interval][1]
+                    comparison_end = end - comparison_interval
+                    start = comparison_end - duration
+                    comparison_results = condition_inst.batch_query(
+                        group_ids=group_ids,
+                        start=start.replace(tzinfo=UTC),
+                        end=comparison_end.replace(tzinfo=UTC),
+                        environment_id=unique_condition.environment_id,
+                    )
+
+                    results = {
+                        group_id: percent_increase(results[group_id], comparison_results[group_id])
+                        for group_id in group_ids
+                    }
+
+        condition_group_results[unique_condition] = results
+
+    # Step 6: For each rule and group applying to that rule, check if the group
+    # meets the conditions of the rule (basically doing BaseEventFrequencyCondition.passes)
+    rules_to_fire = defaultdict(set)
+    # TODO this quad nested for loop sucks
+    for rule in alert_rules:
+        for group_id in rules_to_groups[rule.id]:
+            for rule_condition in rule.data.get(
+                "conditions", []
+            ):  # can this be in it's own mapping? {rule: slow_conditions}
+                for condition, result in condition_group_results.items():
+                    if rule_condition["id"] == condition.cls_id:
+                        interval = rule_condition.get("interval")
+                        match = rule.data.get("filter_match")
```

Comment:
Actually do we even need to worry about the predicates since `apply_rule` is supposed to handle it: https://www.notion.so/sentry/Fix-INC-666-62b150e46f564a3484c7929dcba7de4a?pvs=4#e226b75ca0644e05a1d9edd278780c33 (scroll down to `if predicate_func`)